### DISCUSSION TITLE
Replace the samples-server

### DISCUSSION THREAD

**Discussion by wbamberg** [mdn/mdn-community#496] [Content]
MDN had a custom server, https://github.com/mdn/samples-server. According to the README it's for for live samples that need "a less restrictive environment than running in an iframe on Kuma" - most particularly I think for samples that need some custom server-side handling.

We've decommissioned the sample server, so we need to find a replacement.

I believe this server only served four examples that are used on MDN:

https://mdn-samples.mozilla.org/snippets/html/iframe-simple-contents.html
https://mdn-samples.mozilla.org/s/webrtc-capturestill
https://mdn-samples.mozilla.org/s/webrtc-simple-datachannel
https://mdn-samples.mozilla.org/s/websocket-chat

...referenced from the following MDN pages:

https://developer.mozilla.org/en-us/web/html/element/iframe/
https://developer.mozilla.org/en-us/web/api/webrtc_api/taking_still_photos/
https://developer.mozilla.org/en-us/web/api/webrtc_api/simple_rtcdatachannel_sample/
https://developer.mozilla.org/en-us/web/api/websockets_api/writing_websocket_client_applications/

Since there are only a very few pages involved, it would be good to do this cheaply: to avoid making this a huge project or one that gives the dev team an lot of extra code to maintain.

The most urgent example is probably the iframe one referenced from https://developer.mozilla.org/en-us/web/html/element/iframe/ - someone already filed an issue about this: https://github.com/mdn/content/issues/4116. This is also probably the easiest to fix, since it doesn't require a custom server-side component.

One suggestion is to use (something like) Glitch, and it would probably be worth trying this out.

**Comment by Gregoor** [comment]
An idea I had yesterday while the meeting was already coming to an end: We could also publish these node apps and instead of linking to a server we have to maintain (and/or pay for), we could ask people to run something like:
```sh
npx @mdn/sample-server websocket-chat
```
Which could then also open the site in their default browser.

A downside: It would burden the people with having to have installed Node beforehand.

**Reply by wbamberg** [reply]
Is the suggestion that normal MDN readers should do this, just to use our examples? I think that is too much to ask.

**Comment by wbamberg** [comment]
See also https://github.com/mdn/content/issues/5389 in which it looks like someone used Glitch for a similar use case, and the account has stopped working.

**Comment by hamishwillee** [comment]
Has there been any more thought on this? 

I was looking at how I might host another WebRTC example for WebRTC Encoded Transforms. I was going to use the approach in [Using DTMF](https://developer.mozilla.org/en-US/docs/Web/API/WebRTC_API/Using_DTMF) of faking the server (which you can do because you control both ends of the peer connection). However I'm blocked because an encoded transform runs in a worker, and MDN won't allow that to load inline ... or offline either due to CSP restrictions.

**Reply by bsmth** [reply]
Is having the code in a repository like `mdn/webrtc-examples` acceptable? The fact that these need special handling might indicate the skill level / prerequisite knowledge needed, so being able to run `./start_server.sh` or `python3 start.py` or `npm start` might be okay in these cases, what do you reckon?

__edit:__ in other words, maybe we don't need an additional server if we can assume that readers should be comfortable with running the examples themselves if they would like to see them in action.

**Reply by hamishwillee** [reply]
>  maybe we don't need an additional server if we can assume that readers should be comfortable with running the examples themselves if they would like to see them in action.

@wbamberg comment was "I think that is too much to ask.". 

This issue has sat around for more than 2 years, in which time we've provided no options. Better to provide an option and perhaps one day solve this properly? I.e. I agree with Will, but don't see us every fixing this.

**Reply by wbamberg** [reply]
I guess I made this comment >2 years ago, when it seemed like this kind of platform issue might get addressed :(. I'm not sure it's really about skill level as much as it's about what a pain it is to have to go through these hoops to run an example, and who will really bother to do that, and how we will support them when things go wrong. I suppose it is better than nothing, but perhaps only just - and it's not a zero-cost solution either.

**Reply by bsmth** [reply]
What are the requirements for the server? Is it something that [gh-pages cannot handle](https://github.com/webrtc/samples)?

**Reply by hamishwillee** [reply]
> Is it something that [gh-pages cannot handle](https://github.com/webrtc/samples)?

WebRTC is mediated by a separate server that connects the two ends of a peer connection. I haven't checked these examples, but they should require such a server to be "much use". So no.

**Reply by hamishwillee** [reply]
Or to put it another way, I "think" we need infrastructure to run our own server.
